avr-hal-generic/src/i2c.rs

```diff
@@ -499,5 +499,82 @@ macro_rules! impl_twi_i2c {
                 Ok(())
             }
         }
+
+        $(#[$i2c_attr])*
+        pub struct Slave<CLOCK: $crate::clock::Clock, M> {
+            p: $I2C,
+            _clock: ::core::marker::PhantomData<CLOCK>,
+            sda: $sdamod::$SDA<M>,
+            scl: $sclmod::$SCL<M>,
+            address: u8,
+        }
+
+        impl<CLOCK> Slave<CLOCK, $crate::i2c::I2cPullUp>
+        where
+            CLOCK: $crate::clock::Clock,
+        {
+            /// Initialize the I2C bus
+            ///
+            /// `new()` will enable the internal pull-ups to comply with the I2C
+            /// specification.  If you have external pull-ups connected, please
+            /// use `new_with_external_pullup()` instead.
+            ///
+            /// TODO: verify whether we need the speed param (bitrate)
+            pub fn new(
+                p: $I2C,
+                sda: $sdamod::$SDA<$crate::port::mode::Input<$crate::port::mode::PullUp>>,
+                scl: $sclmod::$SCL<$crate::port::mode::Input<$crate::port::mode::PullUp>>,
+                speed: u32,
+                address: u8,
+            ) -> Slave<CLOCK, $crate::i2c::I2cPullUp> {
+                // Calculate TWBR
+                let twbr = ((CLOCK::FREQ / speed) - 16) / 2;
+                p.$twbr.write(|w| unsafe { w.bits(twbr as u8) });
```

Comment:
I need to research what TWBR does, esp. in the context of an `i2c slave`.

Reply:
TWBR is the `baudrate` or clock speed of the I2c bus.  The slave is not making such decisions as the clock is driven by the master thus I don't think TWBR is used for slave operation (Hate the master/slave teminology here but until a different naming scheme for I2C becomes popular we'll have to stick with it I guess ...).